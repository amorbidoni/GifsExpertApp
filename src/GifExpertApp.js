import React, { useState } from "react";
import AddCategoryApp from "./components/AddCategory";
import GifGridApp from "./components/GifGrid";

const GifExpertApp = () => {
  // no lo hacemos de la siguiente manera a menos que el arr nunca vaya a cambiar, es decir que no vaya a agregar ni sacar categorias y necesite que react lo vuelva a renderizar.
  //   const categories = ["One Ponch", "Samuray X", "Dragon Ball"];
  //
  // en caso de necesitar que se vuelva  a renderizar lo hacemos con el hook useState
  //
  let initialValue = ["One Ponch"];
  const [categories, setCategories] = useState(initialValue);

  // no debemos mutar categories con .push. debemos usar el setCategories
  //   const handleApp = () => {
  //     //    el setCategories reemplaza el valor anterior del useState. por ende usamos el express operator para copiar el arr y agregar una categoria nueva
  //     // setCategories([...categories, "HunterXHunter"]);
  //     // la otra forma es con un callback en el setCategories
  //     setCategories((cats) => [...cats, "HunterXHunter"]);
  //   };

  return (
    <>
      <h2>GifExpertApp</h2>
      <AddCategoryApp setCategories={setCategories} />
      <hr />
      {/* <button onClick={handleApp}>Agregar</button> */}
      <ol>
        {categories.map((category, i) => (
          <GifGridApp key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
