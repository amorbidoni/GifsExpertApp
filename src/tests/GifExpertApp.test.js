import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("pruebas sobre el componente GifExpertApp", () => {
  test("corroborar si se recibe la informacion correcta", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });
  test("debe mostrar una lista de categorias", () => {
    const categories = ["one ponch", "dragon ball"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGridApp").length).toBe(categories.length);
  });
});
