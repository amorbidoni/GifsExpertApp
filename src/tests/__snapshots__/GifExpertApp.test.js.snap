// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas sobre el componente GifExpertApp corroborar si se recibe la informacion correcta 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryApp
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`pruebas sobre el componente GifExpertApp debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoryApp
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGridApp
      category="one ponch"
      key="one ponch"
    />
    <GifGridApp
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
